sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,780 @@
+            device=self._device,
+        )
+
+        train_dataloader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        if not y_test.empty:
+            val_dataset = MomentPytorchDataset(
+                y=y_test,
+                fh=self._model_fh,
+                seq_len=self._seq_len,
+                device=self._device,
+            )
+
+            if self.eval_batch_size == "all":
+                self._eval_batch_size = len(val_dataset)
+            else:
+                self._eval_batch_size = self.eval_batch_size
+
+            val_dataloader = DataLoader(
+                val_dataset, batch_size=self._eval_batch_size, shuffle=True
+            )
+        else:
+            val_dataloader = None
+
+        criterion = self._criterion
+        optimizer = Adam(self.model.parameters(), lr=self.max_lr)
+        # Enable mixed precision training
+
+        # Create a OneCycleLR scheduler
+        total_steps = len(train_dataloader) * max_epoch
+        scheduler = OneCycleLR(
+            optimizer,
+            max_lr=self.max_lr,
+            total_steps=total_steps,
+            pct_start=self.pct_start,
+        )
+
+        # Gradient clipping value
+        max_norm = self.max_norm
+
+        self.model, optimizer, train_dataloader, val_dataloader, scheduler = (
+            accelerator.prepare(
+                self.model, optimizer, train_dataloader, val_dataloader, scheduler
+            )
+        )
+
+        while cur_epoch < max_epoch:
+            cur_epoch = _run_epoch(
+                cur_epoch,
+                accelerator,
+                criterion,
+                optimizer,
+                scheduler,
+                self.model,
+                max_norm,
+                train_dataloader,
+                val_dataloader,
+            )
+
+        if self.return_model_to_cpu:
+            self.model.to("cpu")
+            empty_cache()
+
+        return self
+
+    def _predict(self, y, X=None, fh=None):
+        """Predict method to forecast timesteps into the future.
+
+        fh should not be passed here and
+        must be the same length as the one used to fit the model.
+        """
+        # use y values from fit if y is None in predict
+        if y is None:
+            y = self._y
+
+        index = self._fh.to_absolute_index(self.cutoff)
+        from torch import from_numpy
+
+        self.model = self.model.to(self._device)
+        self.model.eval()
+        y_index_names = list(y.index.names)
+        if isinstance(y.index, pd.MultiIndex):
+            y_ = _frame2numpy(y)
+        else:
+            y_ = np.expand_dims(y.values, axis=0)
+
+        num_instances, sequence_length, num_channels = (
+            y_.shape
+        )  # shape of our input to predict
+        # raise warning if sequence length of y is greater than the sequence
+        # length used to fit the model
+        if sequence_length > self._seq_len:
```

Comment:
I would remove this warning, since it probably will be triggered if you are fitting on a long time series and do not pass `y` to predict. (But it is okay if you think it should stay here).

Reply:
I think `y` is mandatory to be passed in from the user, `momentfm` cannot do zero shot forecasting